### PR TITLE
[WIP] Lazy binary install strategy

### PR DESCRIPTION
**Pull request by waltsims**
This PR should update the install strategy to remove the requirement of pushing compiled binaries to pypi.

### CODE REVIEW

**Comment by waltsims** [comment]
This is related to #86.

It might be the best strategy to lazily install the binaries at runtime... as seen [here](https://stackoverflow.com/a/37553710). Perhaps we check if the binaries are present when we initialize the subpackage kwave and and download the binaries if not.

File: docs/README.md [hunk 1]
```diff
@@ -4,19 +4,27 @@ This project is a Python implementation of most of the [MATLAB toolbox k-Wave](h
...
+
 ```bash
 pip install k-wave-python
 ```
 
-Currently, we are looking for beta testers on Windows.
+After installation, you can download the binaries by running:
 
+```bash
+python install_binaries.py
```
Comment (faridyagubbayli) [inline docs/README.md:23]: Where is this file located?

File: kwave/__init__.py [hunk 2]
```diff
@@ -1,22 +1,123 @@
...
+
+    """
+
+    # GitHub release URLs
+    url_dict = {
+        "linux": {
+            "cuda": [url_base + "kspaceFirstOrder-CUDA-linux/releases/download/v1.3/kspaceFirstOrder-CUDA"],
+            "cpu": [url_base + "kspaceFirstOrder-OMP-linux/releases/download/v1.3.0/kspaceFirstOrder-OMP"],
+        },
+        # "darwin": {
```
Comment (faridyagubbayli) [inline kwave/__init__.py:79]: Consider removing the comments if we don't support lazy-installation in `darwin` yet

File: kwave/executor.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def __init__(self, device):
         if self._is_linux:
             binary_folder = 'linux'
         elif self._is_windows:
-            binary_folder = 'windows'
+            binary_folder = 'cygwin'
```
Comment (faridyagubbayli) [inline kwave/executor.py:31]: Do we support `cygwin` only? If we support different systems in Windows, I would suggest keeping the name `windows` to be more general

**Review by faridyagubbayli — COMMENTED** [review]
Left some questions

**Comment by faridyagubbayli** [comment]
Tests under Windows OS failing because `sys.platform` returns `win32` and we don't have an entry for it. Do you know if we can just say `win32` and `cygwin` are the same thing?

File: docs/README.md (hunk 1 above)
Comment (waltsims) [reply docs/README.md:23]: fixed

**Comment by waltsims** [comment]
TODO: test example locally on linux and windows.

**Comment by waltsims** [comment]
- [ ] Binaries are not downloaded to the site install location but rather the local repository.

**Pull request merged**